resume-buddy/src/main/java/com/google/sps/servlets/BlobstoreServeServlet.java

```diff
@@ -40,13 +40,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     reviewerQuery.setFilter(reviewerFilter);
     PreparedQuery reviewerResults = datastore.prepare(reviewerQuery);
     PreparedQuery revieweeResults = datastore.prepare(revieweeQuery);
-    String matchBlobKeyString, newResumeFileName, showAnnoToolString = "";
+    String matchBlobKeyString = "";
+    String newResumeFileName = "";
+    String showAnnoToolString = "";
+    int revieweeCount = revieweeResults.countEntities(FetchOptions.Builder.withDefaults());
+    int reviewerCount = reviewerResults.countEntities(FetchOptions.Builder.withDefaults());
 
-    if (revieweeResults.countEntities(FetchOptions.Builder.withDefaults()) == 0) {
+    if (revieweeCount == 0 && reviewerCount > 0) {
       matchBlobKeyString = reviewerResults.asSingleEntity().getProperty("resumeBlobKey").toString();
       newResumeFileName = reviewerResults.asSingleEntity().getProperty("reviewee").toString();
       showAnnoTool = true;
-    } else {
+    } else if (revieweeCount > 0 && revieweeCount == 0) {
```

Comment:
This can just be `else if(revieweeCount > 0)` if you first check if they're both 0.